### ISSUE TITLE
Full tone wavelet decomp fails

### ISSUE THREAD

**Issue by jthermiz** [izzy-baldacci/laminar_uecog_viz#4]
Full wavelet decomposition of a full tone datasets craps out on catscan due to limited memory (see error below). I resampled the data from 12.xxxkHz to 4kHz. Possible solutions:

Do wavelet transform for one channel at a time
    -Pros: Should work
    -Cons: Will need to load one channel at a time in memory to do analysis
Get rid of data:
Resample down to 1kHz and only compute wavelet transform up to 500Hz
        -Pros: Should be able to load all channels into memory
        -Cons: Lose high frequency information
Only compute transform on trialize data
        -Pros: Memory efficient
        -Cons: Doesn’t make sense if you want to transform the full trial like for. full tone where trial duration is 200ms?)
Don’t compute the transform for every sample? (especially for low frequencies since those signal change slowly) Or compute for every sample but then throw away samples?
        -Pros: Memory efficient
        -Cons: Requires new implementation/testing

This problem is only going to get worse with Neuropixels. I personally favor 1, 2a and maybe 2b if we are only interested in +/-100 ms of full tone trials. What do folks think? Any other ideas?
    

Traceback (most recent call last):
  File "spectral_test.py", line 57, in <module>
    wavelet_coefs, _, f, _ = wavelet_transform.wavelet_transform(data_proc, fs, hg_only=False)
  File "/home/jhermiz/.conda/envs/nsds_nwb/lib/python3.7/site-packages/process_nwb/wavelet_transform.py", line 144, in wavelet_transform
    Xh = np.zeros(X.shape + (len(filters),), dtype=np.complex)
numpy.core._exceptions.MemoryError: Unable to allocate 603. GiB for an array with shape (5855571, 128, 54) and data type complex128

**Comment by vngutierrez** [comment]
@jthermiz Do you use process_nwb to downsample? Also I agree with 1a and 2a.

**Comment by jthermiz** [comment]
@vngutierrez Yes there's a file called `resample` that we use for throwing away timepoints correctly. I'll work on writing a notebook that processes a dataset in a memory efficient way. I can share it with you afterwards for further development and testing.

**Comment by jthermiz** [comment]
This has been resolved (mostly) due to memory mgmt  improvement with `process_nwb`
